jwst/datamodels/schemas/filteroffset.schema.yaml

```diff
@@ -27,28 +27,14 @@ allOf:
         are matched with the SCI exposure metadata to get
         values for "row_offset" and "column_offset".
 
-        The option with the "name" key is used for
-        MIRI data.
-      anyOf:
-        - type: array
-          items:
-            type: object
-            properties:
-              name:
-                type: string
-              column_offset:
-                type: number
-              row_offset:
-                type: number
-        - type: array
-          items:
-            type: object
-            properties:
-              filter:
-                type: string
-              pupil:
-                type: string
-              row_offset:
-                type: number
-              column_offset:
-                type: number
+      items:
+        type: object
+        properties:
+          filter:
+            type: string
+          pupil:
+            type: string
+          row_offset:
+            type: number
+          column_offset:
+            type: number
```

Comment:
The schema should probably be kept as is, as that way old reference file(s) will still be readable. Delivering new reference files will work using the old schema since they satisfy the 2nd `anyOf` type.

Or is it inconceivable that once the new reffile is delivered, anyone will use the old reffiles for any data?  I.e. will they be removed from rmaps?

Reply:
With the change to use the generic file format, reference files in the old format will be unusable by the master branch and any future tagged releases.  As such, it would only be folks using old builds such as B7.3 that used the old files, for which I believe the old schema would be available anyway?

As such I don't think there's a benefit to keeping the old schema around; indeed, removing it is probably good since anything in that format will be incompatible with the pipeline.

Reply:
Since the `assign_wcs` code is also being updated to match the schema change, then I think @drlaw1558 is correct that new code will only work with new ref files and at the same time old code could be used to work with old ref files (requiring the use of certain CRDS_CONTEXT values with the new and old code as well).

Of course this also means that this PR can't be merged until the new/updated ref files are in place in CRDS, right? Because the new code won't work with the existing ref files.